charts/headlamp/templates/deployment.yaml

```diff
@@ -36,28 +36,28 @@ spec:
           image: "{{ .Values.image.registry}}/{{ .Values.image.repository }}:{{ .Values.image.tag | default (printf "v%s" .Chart.AppVersion) }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           env:
-            {{- with .Values.config.oidc.clientID }}
+            {{- if or .Values.config.oidc.clientID (not .Values.config.oidc.create) }}
```

Comment:
Instead of checking multiple times, how about something like

```helm
{{- $oidcSecretCreated := or .Values.config.oidc.clientID .Values.config.oidc.clientSecret .Values.config.oidc.issuerURL .Values.config.oidc.scopes }}
{{- if or $oidcSecretCreated (not .Values.config.oidc.create) }}
- name: OIDC_CLIENT_ID
  valueFrom:
    secretKeyRef:
      name: oidc
      key: clientID

- name: OIDC_CLIENT_SECRET
  valueFrom:
    secretKeyRef:
      name: oidc
      key: clientSecret

- name: OIDC_ISSUER_URL
  valueFrom:
    secretKeyRef:
      name: oidc
      key: issuerURL

- name: OIDC_SCOPES
  valueFrom:
    secretKeyRef:
      name: oidc
      key: scopes
{{- end }}
```